File: client/src/app/shared/models/blog.ts
export interface Blog {
    id: number;
    title: string;
    blogContent: string;
    picture: string;
    publishedOn: Date;
    updatedOn: Date;
    username: string;
}

export interface BlogCreateEdit {
    id: number;
    title: string;
    blogContent: string;
    picture: File;
}

export class BlogClass {

    constructor(
        public id: number,
        public title: string,
        public content: string,
        public applicationUserId: number,
        public username: string,
        public publishDate: Date,
        public updateDate: Date,
        public deleteConfirm: boolean = false,
        public picture: string
    ) {}

}

export class BlogCreateEditClass {

    constructor(
        public id: number,
        public title: string,
        public content: string,
        public picture: string
    ) {}

}

File: Core/Entities/Blogs/Blog.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.Entities.Blogs
{
    public class Blog : BaseEntity
    {
        public int ApplicationUserId { get; set; }
        public ApplicationUser ApplicationUser { get; set; }
        public string Title { get; set; }
        public string BlogContent { get; set; }
        public string Picture { get; set; }


        [DataType(DataType.Date)]
        public DateTime PublishedOn { get; set; }

        [DataType(DataType.Date)]
        public DateTime UpdatedOn { get; set; }
    }
}